main.go

```diff
@@ -32,85 +34,139 @@ type MyResponse struct {
 	Message string `json:"message"`
 }
 
-func HandleLambdaEvent(event ObjectCreatedEvent) (MyResponse, error) {
-  bucketName := event.Records[0].S3.Bucket.Name
-  objectKey := event.Records[0].S3.Object.Key
+type LambdaTagValues struct {
+	pass string
+	fail string
+}
 
-	sess := session.Must(session.NewSession())
+type DownloaderInterface interface {
+	Download(w io.WriterAt, input *s3.GetObjectInput, options ...func(*s3manager.Downloader)) (n int64, err error)
+}
 
-	endpoint := os.Getenv("AWS_S3_ENDPOINT")
-	sess.Config.Endpoint = &endpoint
-	sess.Config.S3ForcePathStyle = aws.Bool(true)
+type Lambda struct {
+	tagKey     string
+	tagValues  LambdaTagValues
+	s3         s3iface.S3API
+	downloader DownloaderInterface
+}
 
-	downloader := s3manager.NewDownloader(sess)
+func (l Lambda) path() string {
+	return "/tmp/file"
+}
+
+func (l Lambda) downloadFile(bucket string, key string) error {
+	filename := l.path()
 
-	filename := "/tmp/file"
 	f, err := os.Create(filename)
 	if err != nil {
 		log.Printf("failed to create file %q, %v", filename, err)
-		return MyResponse{}, fmt.Errorf("failed to create file %q, %v", filename, err)
+		return fmt.Errorf("failed to create file %q, %v", filename, err)
 	}
 
-	_, err = downloader.Download(f, &s3.GetObjectInput{
-		Bucket: aws.String(bucketName),
-		Key:    aws.String(objectKey),
+	_, err = l.downloader.Download(f, &s3.GetObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
 	})
 
 	if err != nil {
 		log.Printf("failed to download file, %v", err)
-		return MyResponse{}, fmt.Errorf("failed to download file, %v", err)
+		return fmt.Errorf("failed to download file, %v", err)
 	}
 
-	// scan with clamav
-	status := "ok"
-	cmd := exec.Command("./bin/clamscan", filename)
+	return nil
+}
+
+func (l Lambda) scanFile() (string, error) {
```

Comment:
might make more sense to have this not on Lambda, as `func scanFile(path string) (bool, error)` (or create a `Scanner` type if you want to mock it)